synapse/rest/media/v1/media_repository.py

```diff
@@ -87,18 +96,77 @@ def _makedirs(filepath):
         if not os.path.exists(dirname):
             os.makedirs(dirname)
 
+    @staticmethod
+    def _write_file_synchronously(source, fname, close_source=False):
+        source.seek(0)  # Ensure we read from the start of the file
+        with open(fname, "wb") as f:
+            shutil.copyfileobj(source, f)
+
+        if close_source:
+            source.close()
```

Comment:
Yes, it is very sad, but I don't know how else to do it since the writing to the backup can happen asynchronously, and so may last longer than the context which calls it.

Reply:
(The other option is to rely on the GC closing the file, but that doesn't sound great either)

Reply:
I guess we could copy from the file we wrote into the primary media store, but thats a bit sad making in terms of having to reload that file back into memory.